### PR TITLE
Add the pool screens and associated states

### PR DESCRIPTION
**Pull request by alexandrepiveteau**
This PR adds support for the pools screens (see Figma). For convenience, some `@Preview` can be found in this Gist : https://gist.github.com/alexandrepiveteau/4287fbe9f2fc1a5739bfc45b9fec4fb0.

## Deliverables

telegram-cloud-photo-size-4-6034891451810299096-y
telegram-cloud-photo-size-4-6034891451810299095-y

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/PoolCard.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  Text(
+      text = name,
+      modifier = modifier,
+      style = MaterialTheme.typography.subtitle1,
+      color = PawniesColors.Green200,
+  )
+}
+
+/**
+ * Displays the color of the pool.
```
Comment (Fouad-sys) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/PoolCard.kt:114]: [Nitpick] Doesn't this composable function display the status of the pool?
```suggestion
 * Displays the status of the pool.
```

**Review by Fouad-sys — APPROVED** [review]
Very neat and well done screens, Great job!

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/PoolTable.kt [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  fun P.scoreAgainst(other: P): PoolScore?
+}
+
+/**
+ * The table of the players which are participating in a pool.
+ *
+ * @param P the type of the [PoolMember] which participate in the pool.
+ * @param data the [PoolData] which must be displayed.
+ * @param modifier the [Modifier] for this composable.
+ */
```
Comment (matt989253) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/PoolTable.kt:72]: Missing parameter documentation
```suggestion
 * @param textStyle the [TextStyle] for this composable.
 */
```

**Review by matt989253 — COMMENTED** [review]
Great job! Looks amazing!

Concerning the colors not being in MaterialTheme, I think if we wish to do something about it it would require it's own feature with a custom material theme so you can probably ignore my comment for this PR.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/DashedDivider.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * @param interval the size of the intervals.
+ */
+@Composable
+fun DashedDivider(
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    width: Dp = 2.dp,
+    interval: Dp = 4.dp,
+) {
+  val currentColor = color.takeOrElse { PawniesColors.Green100 }
```
[the remark this answers is not in the document]
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/DashedDivider.kt:32]: I agree with you that we should probably have our own custom theme with the right colors for our design system. I've created an issue (#319) to keep track of this problem.

**Review by matt989253 — APPROVED** [review]

**Review by Fouad-sys — APPROVED** [review]
Amazingly done!

**Pull request merged**